quint/src/graphics.ts

```diff
@@ -123,6 +121,99 @@ export function prettyQuintEx(ex: QuintEx): Doc {
   }
 }
 
+export function prettyQuintDef(
+  def: QuintDef,
+  includeBody: boolean = true,
+  type: TypeScheme | undefined = undefined
+): Doc {
+  const typeAnnotation = isAnnotatedDef(def)
+    ? [text(': '), prettyQuintType(def.typeAnnotation)]
+    : type
+    ? [text(': '), prettyTypeScheme(type)]
+    : []
+  switch (def.kind) {
+    case 'def': {
+      const header = group([
+        richtext(chalk.magenta, qualifierToString(def.qualifier)),
+        text(' '),
+        richtext(chalk.blue, def.name),
+        ...typeAnnotation,
+      ])
+      return includeBody ? group([header, text(' = '), prettyQuintEx(def.expr)]) : header
+    }
+    default:
+      // TODO, not used for now
+      return text(definitionToString(def))
+  }
+}
+
+export function prettyTypeScheme(scheme: TypeScheme): Doc {
+  const [vars, type] = canonicalTypeScheme(scheme)
+  const varsDoc =
+    vars.length > 0 ? group([text('∀ '), docJoin([text(','), line()], vars.map(text)), text(' . ')]) : text('')
```

Comment:
I am not sure many people would understand the role of the quantifier here. Would not it be better to use some form of a template-like syntax, e.g., `[x, y, z]. type`?

Reply:
I don't see how people would recognize/understand templates more than universal quantification.

Thinking further, I think we should have two forms of printing signatures as we have to forms of defining signatures in our syntax. The `∀ a . (a) => a` like form should be used for adopters of the haskell-like signatures, while for the rest, we could try to mimic rust/java/ts-like generics: `foo<a>(param: a): a`

Reply:
I've opened an [issue](https://github.com/informalsystems/quint/issues/901) to discuss this further. I'll not block this PR on your comment as this was not introduced by the PR itself.